dbt/node_runners.py

```diff
@@ -348,6 +350,8 @@ def run_hooks(cls, config, adapter, manifest, hook_type, extra_context):
                 dbt.contracts.graph.parsed.Hook(**hook_dict)
 
             sql = hook_dict.get('sql', '')
+            dbt.ui.printer.print_timestamped_line("{} of {} {} ".
+                                                  format(i+1, hook_count, sql))
```

Comment:
you can construct a `RunModelResult` object here with something like:
```
res = RunModelResult(hook, status=status)
```

You can get a status from the `adapter.execute` call below. That will return a tuple of a status and an Agate dataframe, as shown here: https://github.com/fishtown-analytics/dbt/blob/dev/stephen-girard/dbt/adapters/base/impl.py#L138

eg:
```
status, _ = adapter.execute(sql, model_name=model_name, auto_begin=False, fetch=False)
```

Reply:
^ I think this will work, but I haven't tested it. Let me know if you hit any more snags